packages/cli/src/commands/styles/utils/publish.main.ts

```diff
@@ -0,0 +1,172 @@
+//validator //loader/parse //tokenizer
+import axios from 'axios';
+import { StyleMetadata } from 'figma-api/lib/api-types';
+import { DesignToken, processTypographyToken, NodeDocument } from './figma.utils';
+import { groupBy } from './common.utils';
+
+// const figmaToken = '379431-5a32d6d8-85b0-4193-b06b-841b89dcf741';
+export type StyleResponse = {
+  meta: {
+    styles: StyleDescriptor[],
+  },
+};
+export type StyleDescriptor = {
+  key: string,
+  file_key: string,
+  node_id: string,
+  style_type: string,
+  thumbnail_url: string,
+  name: string,
+  description: string,
+  created_at: string,
+  updated_at: string,
+  user: User,
+  sort_position: string,
+};
+type User = {
+  id: string,
+  handle: string,
+  img_url: string,
+};
+
+type GetStylesListType = {
+  nodes: string[],
+  styles: {
+    [key: string]: StyleMetadata,
+  },
+};
+
+const groupByType = <T extends DesignToken>(list: T[]) => groupBy(list, 'type');
+
+export const getFileKey = (url: string) => {
+  const fileKey = url.match(/\/file\/(\w*)\//);
+  if(fileKey && fileKey.length > 1) return fileKey[1];
+  throw Error('Could not find the file URL in the figma file');
+};
+
+export const generateToken = (name: string) => {
+  return `--${ name.toLowerCase().split('/').join('-').split(' ').join('-') }`;
+};
+
+const hex = (n: number) => `00${ n.toString(16) }`.slice(-2);
+export const colorToHex = ({ r, g, b }: any) =>
+  `#${ [r, g, b]
+    .map((rValue) => rValue * 255)
+    .map(Math.round)
+    .map(hex)
+    .join('') }`;
+
+
+export const parseGRID = (style: StyleDescriptor,target: NodeDocument) => {
+  const out: DesignToken = {
+    type: 'breakpoint',
+    name: target.name,
+    token: generateToken(target.name),
+    value: 'default'
+  };
+  if(style.description.includes('--')){
```

Comment:
is there any reason why we're using regex so heavily here? We'll be working with the designers to come up with the right naming. Unsure how easy this is to maintain